### ISSUE TITLE
Is it feasible to run max of 1 proc at a time per agent?

### ISSUE THREAD

**Issue by gusostow** [alexhsamuel/procstar#41]
I'm kicking around the idea of using procstar as the basis for a containerized parallel computing tool.

**Comment by gusostow** [comment]
Sorry hit enter before I finished typing. Will edit

**Comment by alexhsamuel** [comment]
Yes, the Procstar server tracks processes and connections (and which proc is running on which connection).  It's not guaranteed to be perfectly timely and correct, but it should be good enough for this purpose.  (Mostly, if the proc just completed, this will be briefly out of date, but that's probably OK.)

```
conn_ids = { c.conn_id for c in server.connections }
free_conn_ids = conn_ids - { p.conn_id for p in saerver.processes }
```

Of course you could partition by group ID, etc.

Regards,
Alex

**Comment by alexhsamuel** [comment]
Yes, the Procstar server tracks processes and connections (and which proc is running on which connection).  It's not guaranteed to be perfectly timely and correct, but it should be good enough for this purpose.  (Mostly, if the proc just completed, this will be briefly out of date, but that's probably OK.)

```
conn_ids = { c.conn_id for c in server.connections }
free_conn_ids = conn_ids - { p.conn_id for p in server.processes }
```

Of course you could partition by group ID, etc.
